x-pack/plugins/cases/public/components/case_view/components/user_list.tsx

```diff
@@ -32,55 +34,41 @@ interface UserListProps {
   headline: string;
   loading?: boolean;
   users: ElasticUser[];
+  userProfiles?: Map<string, UserProfileWithAvatar>;
   dataTestSubj?: string;
 }
 
-const MyAvatar = styled(EuiAvatar)`
-  top: -4px;
-`;
-
 const MyFlexGroup = styled(EuiFlexGroup)`
   ${({ theme }) => css`
     margin-top: ${theme.eui.euiSizeM};
   `}
 `;
 
 const renderUsers = (
-  users: ElasticUser[],
+  users: UserInfoWithAvatar[],
   handleSendEmail: (emailAddress: string | undefined | null) => void
 ) =>
-  users.map(({ fullName, username, email }, key) => (
+  users.map((userInfo, key) => (
     <MyFlexGroup key={key} justifyContent="spaceBetween" responsive={false}>
       <EuiFlexItem grow={false}>
-        <EuiFlexGroup gutterSize="xs" responsive={false}>
-          <EuiFlexItem grow={false}>
-            <MyAvatar name={fullName ? fullName : username ?? ''} />
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiToolTip position="top" content={<p>{fullName ? fullName : username ?? ''}</p>}>
-              <p>
-                <strong>
-                  <small data-test-subj="case-view-username">{username}</small>
-                </strong>
-              </p>
-            </EuiToolTip>
-          </EuiFlexItem>
-        </EuiFlexGroup>
+        <HoverableUserWithAvatar userInfo={userInfo} />
       </EuiFlexItem>
       <EuiFlexItem grow={false}>
         <EuiButtonIcon
           data-test-subj="user-list-email-button"
-          onClick={handleSendEmail.bind(null, email)}
+          onClick={handleSendEmail.bind(null, userInfo.user?.email)}
           iconType="email"
-          aria-label={i18n.SEND_EMAIL_ARIA(fullName ? fullName : username ?? '')}
-          isDisabled={isEmpty(email)}
+          aria-label={i18n.SEND_EMAIL_ARIA(
+            userInfo.user?.full_name ? userInfo.user?.full_name : userInfo.user?.username ?? ''
```

Comment:
If I remember correctly we have a utility function to return us a valid name, correct?

Reply:
There's one slight difference, the utility function will do full_name -> email -> username -> unknown. This logic does full_name -> username -> ''. Using the utility function should be fine though right?